src/v/redpanda/application.cc

```diff
@@ -323,8 +328,14 @@ int application::run(int ac, char** av) {
                 hydrate_config(cfg);
                 initialize();
                 check_environment();
+                check_for_crash_loop();
                 setup_metrics();
                 wire_up_and_start(app_signal);
+                // We schedule the deletion _after_ the application fully
+                // starts up. This ensures that any errors like
```

Comment:
> but this will make it quite common for users to get into this situation if they make a bad configuration (and they'll need to know how to skip the loop blocking).

This is what prompted me to turn this off by default. I feel those users who use RP as a Kafka replacement (like in a dev environment and such) may not know enough to look into the log and fix the problem and be stuck in this misconfiguration loop. wdyt. We can probably set this internally in our cloud cluster? 


> I think this could motivate us to also include a configuration checksum in the persistent crash record, so that we can automatically allow the cluster to proceed after its configuration is changed (assuming that the config change is an attempt to fix the issue)

I added a mechanism that resets tracking if node config file is updated.